ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/InsertsPublisher.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client;
+
+import java.util.ArrayDeque;
+import java.util.Objects;
+import java.util.Queue;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * A {@code org.reactivestreams.Publisher} suitable for use with the
+ * {@link Client#streamInserts(String, Publisher)} method. Rows for insertion are passed to the
+ * publisher via the {@link #accept(KsqlObject)} method, and buffered for delivery once the
+ * {@link Client#streamInserts} request is made and the server-side subscriber has been subscribed.
+ */
+public class InsertsPublisher implements Publisher<KsqlObject> {
```

Comment:
I thought when we first discussed providing InsertsPublisher and RowSubscriber (I plan to add this in a future PR) implementations for use with the client, we said the implementations shouldn't be tied to Vert.x. Did I misunderstand what you meant at the time?

Besides not having Vert.x, this InsertsPublisher is also simpler than BufferedPublisher in that it doesn't expose sending custom errors to the subscriber (since there's no reason to do that here), and a few other minor bits.

Also, by decoupling InsertsPublisher from BufferedPublisher, we can evolve BufferedPublisher in breaking ways without affecting the user-facing API.

WDYT?